analyzers/src/SonarAnalyzer.CSharp/Rules/SillyBitwiseOperation.cs

```diff
@@ -64,5 +84,99 @@ private void CheckBinary(SonarSyntaxNodeReportingContext context, int constValue
             var binary = (BinaryExpressionSyntax)context.Node;
             CheckBinary(context, binary.Left, binary.OperatorToken, binary.Right, constValueToLookFor);
         }
+
+        private void CheckComparison(SonarSyntaxNodeReportingContext context)
```

Comment:
Hey @martin-strecker-sonarsource , do you think this approach is better?

Also, maybe we should reconsider the fact that we added this rule as part of `SillyBitwise`, after implementing it I feel like the overlap they have does not justify the cognitive load of having both of them in the same files.

Also, this rule will exist only for C# (at least for a while). I think it would be better if I moved it to its own class. What do you think?

Reply:
Let's decide when the dust has settled. We need to do some work here anyway.